unicode_hostname.go

```diff
@@ -589,6 +589,7 @@ func ZeroWidthNonJoiner(val []rune, idx int) (ok bool) {
 // verifyRFC5891_4_2_3_1 ensures there are no leading, terminating, or double-hypens in the label
 func verifyRFC5891_4_2_3_1(label string) error {
 	var p, q rune
+	var i int
```

Comment:
nit: I think you can only declare `var q rune` here, and in the loop use `for i, p := range runeslice {`. There doesn't appear to be any reason to scope them outside of the loop, and can just complicate things in the future